Buildings/Media/Steam.mo

```diff
@@ -0,0 +1,855 @@
+</p>
+<pre>
+   state := <strong>smooth</strong>(1, <strong>if</strong> x &gt;  x_small <strong>then</strong> state_a <strong>else</strong>
+                      <strong>if</strong> x &lt; -x_small <strong>then</strong> state_b <strong>else</strong> f(state_a, state_b));
+</pre>
+<p>
+This is performed by applying function <strong>Media.Common.smoothStep</strong>(..)
+on every element of the thermodynamic state record.
+</p>
+<p>
+If <strong>mass fractions</strong> X[:] are approximated with this function then this can be performed
+for all <strong>nX</strong> mass fractions, instead of applying it for nX-1 mass fractions and computing
+the last one by the mass fraction constraint sum(X)=1. The reason is that the approximating function has the
+property that sum(state.X) = 1, provided sum(state_a.X) = sum(state_b.X) = 1.
+This can be shown by evaluating the approximating function in the abs(x) &lt; x_small
+region (otherwise state.X is either state_a.X or state_b.X):
+</p>
+<pre>
+    X[1]  = smoothStep(x, X_a[1] , X_b[1] , x_small);
+    X[2]  = smoothStep(x, X_a[2] , X_b[2] , x_small);
+       ...
+    X[nX] = smoothStep(x, X_a[nX], X_b[nX], x_small);
+</pre>
+<p>
+or
+</p>
+<pre>
+    X[1]  = c*(X_a[1]  - X_b[1])  + (X_a[1]  + X_b[1])/2
+    X[2]  = c*(X_a[2]  - X_b[2])  + (X_a[2]  + X_b[2])/2;
+       ...
+    X[nX] = c*(X_a[nX] - X_b[nX]) + (X_a[nX] + X_b[nX])/2;
+    c     = (x/x_small)*((x/x_small)^2 - 3)/4
+</pre>
+<p>
+Summing all mass fractions together results in
+</p>
+<pre>
+    sum(X) = c*(sum(X_a) - sum(X_b)) + (sum(X_a) + sum(X_b))/2
+           = c*(1 - 1) + (1 + 1)/2
+           = 1
+</pre>
+</html>"));
+  end setSmoothState;
+
+  redeclare replaceable function extends saturationState
+    "Return saturation property record from pressure"
+  algorithm
+    sat.psat := p;
+    sat.Tsat := saturationTemperature(p);
+    annotation (
+      smoothOrder=2,
+      Documentation(
+        info="<html>
+    <p>
+    Returns the saturation state for given pressure. This relation is
+    valid in the region of <i>0</i> to <i>800</i> C (<i>0</i> to <i>100</i> MPa).
+    This corresponds to Region 2 of the IAPWS-IF97 water medium models.
+    </p>
+</html>",
+        revisions="<html>
+<ul>
+<li>
+May 6, 2020, by Kathryn Hinkelman:<br/>
+First implementation.
+</li>
+</ul>
+</html>"));
+  end saturationState;
+//////////////////////////////////////////////////////////////////////
+// Protected classes.
+// These classes are only of use within this medium model.
+// Models generally have no need to access them.
+// Therefore, they are made protected. This also allows to redeclare the
+// medium model with another medium model that does not provide an
+// implementation of these classes.
+protected
+  function rho_pT
+    "Density as function or pressure and temperature"
+    extends Modelica.Icons.Function;
+    input AbsolutePressure p "Pressure";
+    input Temperature T "Temperature";
+    output Density rho "Density";
+  protected
```

Comment:
I see no need in repeating the "protected".

Reply:
The second `protected` here is within the function. The IF97 models in the MSL protected these parameters since they shouldn't be accessible elsewhere, even via function calls within the media model. I'd prefer to keep them in this case, if you agree?

Reply:
Sounds good to me.